esp_jpeg/jpeg_decoder.c

```diff
@@ -79,13 +79,14 @@ esp_err_t esp_jpeg_decode(esp_jpeg_image_cfg_t *cfg, esp_jpeg_image_output_t *im
     assert(cfg != NULL);
     assert(img != NULL);
 
-    workbuf = heap_caps_malloc(JPEG_WORK_BUF_SIZE, MALLOC_CAP_DEFAULT);
+    const size_t workbuf_size = cfg->advanced.working_buffer_size ? cfg->advanced.working_buffer_size : JPEG_WORK_BUF_SIZE;
+    workbuf = heap_caps_malloc(workbuf_size, MALLOC_CAP_DEFAULT);
```

Comment:
Just an idea, you can also add `alloc_caps` next to `working_buffer_size`, so that the user can easily move the buffer to PSRAM if needed.

Or, `working_buffer_ptr` — which, if set, will be used instead of allocating the new buffer.

Reply:
Thank you for the review @igrr , I'll implement it

I was also thinking about refactoring the memory allocation pool in Tjpgd. So we use dynamic allocation instead of this pre-allocated memory pool... maybe in next release :)